tests/ingress/ingress_test.go

```diff
@@ -0,0 +1,189 @@
+package ingress
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/giantswarm/apptest"
+	"github.com/giantswarm/backoff"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	capzv1alpha3 "sigs.k8s.io/cluster-api-provider-azure/api/v1alpha3"
+	expcapzv1alpha3 "sigs.k8s.io/cluster-api-provider-azure/exp/api/v1alpha3"
+	capiv1alpha3 "sigs.k8s.io/cluster-api/api/v1alpha3"
+	expcapiv1alpha3 "sigs.k8s.io/cluster-api/exp/api/v1alpha3"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/giantswarm/azure-sonobuoy/v5/tests/ctrlclient"
+)
+
+const (
+	helloWorldAppName = "loadtest-app"
+	IngressValues     = `baseDomain: %s
+`
+	HelloWorldValues = `replicaCount: 1
+ingress:
+  enabled: true
+  annotations:
+    kubernetes.io/ingress.class: nginx
+  paths:
+    - "/"
+  hosts:
+    - %s
+`
+)
+
+func Test_Ingress(t *testing.T) {
+	var err error
+
+	ctx := context.Background()
+
+	cpCtrlClient, err := ctrlclient.CreateCPCtrlClient(ctx)
+	if err != nil {
+		t.Fatalf("error creating CP k8s client: %v", err)
+	}
+
+	logger, err := micrologger.New(micrologger.Config{})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	clusterID, exists := os.LookupEnv("CLUSTER_ID")
+	if !exists {
+		t.Fatal("missing CLUSTER_ID environment variable")
+	}
+
+	cpKubeConfig, err := ctrlclient.GetCPKubeConfig(ctx)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	tcKubeConfig, err := ctrlclient.GetTCKubeConfig(ctx)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	clusterList := &capiv1alpha3.ClusterList{}
+	err = cpCtrlClient.List(ctx, clusterList, client.MatchingLabels{capiv1alpha3.ClusterLabelName: clusterID})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	baseDomain := strings.TrimPrefix(clusterList.Items[0].Spec.ControlPlaneEndpoint.Host, "api.")
+	appEndpoint := fmt.Sprintf("%s.%s", helloWorldAppName, baseDomain)
+
+	var appTest apptest.Interface
+	{
+		runtimeScheme := runtime.NewScheme()
+		appSchemeBuilder := runtime.SchemeBuilder{
+			capiv1alpha3.AddToScheme,
+			capzv1alpha3.AddToScheme,
+			expcapiv1alpha3.AddToScheme,
+			expcapzv1alpha3.AddToScheme,
+			corev1.AddToScheme,
+		}
```

Comment:
I feel like we declare these things all over the place. Would it make sense to create something like this:
filename: `pkg/scheme/scheme.go`:
```go

var Everything = runtime.NewScheme()

func init() {
    schemeBuilder := runtime.SchemeBuilder{
        capiv1alpha3.AddToScheme,
        capzv1alpha3.AddToScheme,
        expcapiv1alpha3.AddToScheme,
        expcapzv1alpha3.AddToScheme,
        corev1.AddToScheme,
    }
    err := schemeBuilder.AddToScheme(Everything)
    if err != nil {
        panic(err)
    }
}
```

So we could then just use `scheme.Everything` in our code.

Alternatively we could scope it a bit and have `pkg/project/scheme.go` with `project.Scheme` that works for current project.

Reply:
Ah, yeah, my thinking with the `pkg/scheme` package was that it could be copied from one project to another, but alternatively it might make more sense to specify the pattern so that it's defined in `pkg/project` package as project specific one so the scope is more suited to corresponding project instead of having all the schemes everywhere.